main.tf

```diff
@@ -2709,15 +2709,69 @@ module "vpa" {
   tags = var.tags
 }
 
-#-----------------Kubernetes Add-ons----------------------
 
-module "fargate_fluentbit" {
-  count         = var.enable_fargate_fluentbit ? 1 : 0
-  source        = "./modules/fargate-fluentbit"
-  addon_config  = var.fargate_fluentbit_addon_config
-  addon_context = local.addon_context
+################################################################################
+# Fargate Fluentbit
+################################################################################
+# Help on Fargate Logging with Fluentbit and CloudWatch
+# https://docs.aws.amazon.com/eks/latest/userguide/fargate-logging.html
+resource "kubernetes_namespace_v1" "aws_observability" {
+  count = var.enable_fargate_fluentbit ? 1 : 0
+  metadata {
+    name = "aws-observability"
+
+    labels = {
+      aws-observability = "enabled"
+    }
+  }
 }
 
+# fluent-bit-cloudwatch value as the name of the CloudWatch log group that is automatically created as soon as your apps start logging
+resource "kubernetes_config_map_v1" "aws_logging" {
+  count = var.enable_fargate_fluentbit ? 1 : 0
+  metadata {
+    name      = "aws-logging"
+    namespace = kubernetes_namespace_v1.aws_observability[0].id
+  }
+
+  data = {
+    "parsers.conf" = try(var.fargate_fluentbit.parsers_conf, <<-EOT
+    [PARSER]
+      Name regex
+      Format regex
+      Regex ^(?<time>[^ ]+) (?<stream>[^ ]+) (?<logtag>[^ ]+) (?<message>.+)$
+      Time_Key time
+      Time_Format %Y-%m-%dT%H:%M:%S.%L%z
+      Time_Keep On
+      Decode_Field_As json message
+    EOT
+    )
+    "filters.conf" = try(var.fargate_fluentbit.filters_conf, <<-EOT
+      [FILTER]
+      Name parser
+      Match *
+      Key_Name log
+      Parser regex
+      Preserve_Key True
+      Reserve_Data True
+    EOT
+    )
+    "output.conf" = try(var.fargate_fluentbit.output_conf, <<-EOT
+    [OUTPUT]
+      Name cloudwatch_logs
+      Match *
+      region ${local.region}
+      log_group_name ${try(var.fargate_fluentbit.cwlog_group, "/${var.cluster_name}/fargate-fluentbit-logs")}
```

Comment:
should we create the cloudwatch log group like the other FluentBit addon? That would give users IaC control of the log group (tag, retention period, encryption, etc.)

Reply:
or at least offer to create the log group (provide the means to create it here)

Reply:
Agree. I'll add the resource creation with a condition, same as the FluentBit.

Reply:
Done!